welcome_wizard/views.py

```diff
@@ -48,140 +58,182 @@ def check_sync(instance, request):
         enqueue_pull_git_repository_and_refresh_data(repo, request.user)
 
 
-class ManufacturerListView(generic.ObjectListView):
-    """Table of all Manufacturers discovered in the Git Repository."""
+class ManufacturerImportUIViewSet(ObjectDetailViewMixin, ObjectListViewMixin, ObjectChangeLogViewMixin):  # pylint: disable=abstract-method
+    """List view for ManufacturerImport."""
 
     permission_required = "welcome_wizard.view_manufacturerimport"
```

Comment:
I don't believe this works on ui viewsets. You'll also need to implement custom permissions for the custom actions below. Take a look at https://github.com/nautobot/nautobot/blob/ccc7fab85b16b98bbd5b0a39d657ddb45222c988/nautobot/dcim/views.py#L1125-L1140